### PR TITLE
BLD: fix build warnings

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref #25995, #17936

AFAICT these warnings are caused by cython casting `len(some_ndarray)` to `int` or `Py_ssize_t` but casting `len(some_memoryview)` to `size_t`.  Easiest fix is to declare the affected len variables as `size_t` since we know they'll never be negative.

### CODE REVIEW

**Comment by WillAyd** [comment]
Hmm understood that these would never be negative but since Py_ssize_t is what Python would use for indexing that still seems like the more natural data type to use. Is there not a way to silence these warnings while still using that type for indexers?

**Comment by jbrockmendel** [comment]
@scoder any thoughts on whether this is worthwhile and if not, how to silence the warnings?

**Comment by scoder** [comment]
`size_t`, or, more generally, unsigned integer types in C, have their advantages and disadvantages. Special use cases like bit manipulation aside, it's nice that they explicitly state that their value cannot be lower than zero. On the downside, they mix badly with C's default (signed) integer behaviour and signed arithmetic in general, and fail to support Python's negative indexing.

I think `Py_ssize_t` is the most obvious return type for `len()`, and memory views shouldn't diverge from this. This is something we can change in Cython 3.0. If you have an opinion, please discuss it at
https://github.com/cython/cython/issues/2992

As for a work-around for now, it's probably best to explicitly cast the return values to `Py_ssize_t`, at least where it's not used in an unsigned context anyway.

**Review by jreback — CHANGES_REQUESTED** [review]
we are consistently using Py_ssize_t, adding in size_t is confusing.

**Comment by jbrockmendel** [comment]
Updated to use Py_ssize_t consistently

File: pandas/_libs/ops.pyx [hunk 1]
```diff
@@ -220,7 +220,7 @@ def vec_binop(object[:] left, object[:] right, object op):
     result : ndarray[object]
     """
     cdef:
-        Py_ssize_t i, n = len(left)
+        size_t i, n = len(left)
```
Comment (jreback) [inline pandas/_libs/ops.pyx]: does this one need updating?

Comment (jbrockmendel) [reply pandas/_libs/ops.pyx]: Yes, fixed.

File: pandas/_libs/tslibs/parsing.pyx [hunk 2]
```diff
@@ -8,6 +8,7 @@ from io import StringIO
 from libc.string cimport strchr
 
 import cython
+from cython import size_t, Py_ssize_t
```
Comment (jreback) [inline pandas/_libs/tslibs/parsing.pyx]: does size_t still need to be imported?

**Review by jreback — CHANGES_REQUESTED** [review]
import comment, otherwise lgtm. ping on green.

File: pandas/_libs/tslibs/parsing.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/parsing.pyx]: Nope,

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. merge on green (and close the issues above if this & others cover)

**Comment by jbrockmendel** [comment]
> and close the issues above if this & others cover

There's still a bunch in interval.pyx that I'm not sure are purely cosmetic (cc @jschendel)

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**